spec/unit/knife/winrm_spec.rb

```diff
@@ -491,4 +491,65 @@
       end
     end
   end
+
+  context "Impact of concurrency value when target nodes are 3" do
+    let(:winrm_user) { 'testuser' }
+    let(:transport) { 'plaintext' }
+    let(:password) { 'testpassword' }
+    let(:protocol) { 'basic' }
+    let(:knife_args) do
+      [
+        '-m', 'localhost knownhost somehost',
+        '-x', winrm_user,
+        '-P', password,
+        '-w', transport,
+        '--winrm-authentication-protocol', protocol,
+        'echo helloworld'
+      ]
+    end
+    let(:winrm_connection) { Dummy::Connection.new }
+
+    subject { Chef::Knife::Winrm.new(knife_args) }
+
+    it "concurrency not set" do
+      expect(subject).to receive(:run_command_in_thread).thrice
```

Comment:
I think there's value in verifying that we've called run_command_in_thread three times, but we also want to verify the number of threads spawned in these cases. Could we do that via something like `expect(Thread).to receive(:new).exactly(n).times.and_call_original` ? That would validate that we're running the command for all hosts, and that the we are not spawning more (or fewer) threads than we should be in each case.

Reply:
Yes I got your point, But when we tried to check number of spawning threads, it actually goes to make connection for all host under `-m` i.e `localhost knownhost somehost` and it fails to connect these dummy host and return a error. and mocking this will not help to get no of spawning thread.

Reply:
I think we could get around that by continuing to check for receive `receive(:run_command_in_thread).thrice` as well; the thread would get spawned, but `run_command_in_thread` would be prevented from execution by the condition.

Reply:
Thank you @marcparadise, Added test cases of `Threads` and rebase it.